sympy/solvers/solvers.py

```diff
@@ -1020,8 +1020,13 @@ def _sympified_list(w):
         # Abs
         fi = fi.replace(Abs, lambda arg:
             separatevars(Abs(arg)) if arg.has(*symbols) else Abs(arg))
-        fi = fi.replace(Abs, lambda arg:
-            Abs(arg).rewrite(Piecewise) if arg.has(*symbols) else Abs(arg))
+        while True:
+            was = fi
+            fi = fi.replace(Abs, lambda arg:
+                (Abs(arg).rewrite(Piecewise) if arg.has(*symbols)
+                else Abs(arg)))
+            if was == fi:
+                break
```

Comment:
Can we restructure this piece of code such that, `while True` is replaced by `while condition`, where condition is actually a condition, rather than a boolean?

Reply:
I think this is fine. It's just a limitation of Python that there are no do/while loops so you end up needing this kind of while/break construct.

Can these lines be combined with those above though?

Reply:
Maybe we should use separatevars here and remove the two lines above. What difference would that make?

Reply:
Can you please elaborate as to what to do exactly, which lines to remove and what to use instead? I will give it a try, if it works fine, we will go ahead.

Reply:
The code currently calls replace twice with almost the same arguments. The difference is the use of separatevars. Your change here makes the second call happen repeatedly in a loop until everything is replaced. Maybe we could do that using separatevars each time in the loop and then maybe we wouldn't need the two lines above that call replace with separatevars.

Reply:
Well, @oscarbenjamin 's idea sounds good. Have you given it a try @sachin-4099 ?

Reply:
Yeah it is working.

Reply:
Can you please push the changes here?

Reply:
Done